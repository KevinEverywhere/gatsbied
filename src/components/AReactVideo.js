import React from 'react'
import PropTypes from 'prop-types'

const AReactVideo = (props) => {
  const radiusStr = `primitive: sphere; radius:${props.radius}`
  if(props.spherical){
    return(
    <a-entity geometry={radiusStr} position={props.position} rotation={props.rotation}>
    <a-video material="side:double" geometry={radiusStr} src={props.src} controls video-texture position="0 0 0" />
    </a-entity>
  )
  }else{
    return(<a-entity position={props.position} rotation={props.rotation}>
      <a-video  src={props.src} controls video-texture width={props.width} height={props.height} position="0 0 0" />
    </a-entity>)
  }
}

AReactVideo.propTypes={
  position: PropTypes.string,
  rotation: PropTypes.string,
  radius: PropTypes.string,
  src: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
}

export default AReactVideo
