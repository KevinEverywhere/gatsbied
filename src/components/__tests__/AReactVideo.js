import React from "react"
import renderer from "react-test-renderer"
import AReactVideo from "../AReactVideo"

describe("AReactVideo", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<AReactVideo />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
