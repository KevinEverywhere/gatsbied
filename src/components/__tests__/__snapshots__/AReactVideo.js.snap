// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AReactVideo renders correctly 1`] = `
<a-entity>
  <a-video
    controls={true}
    position="0 0 0"
    video-texture={true}
  />
</a-entity>
`;
